lib/internal/crypto/cipher.js

```diff
@@ -158,12 +173,21 @@ Cipher.prototype.setAuthTag = function setAuthTag(tagbuf) {
   return this;
 };
 
-Cipher.prototype.setAAD = function setAAD(aadbuf) {
+Cipher.prototype.setAAD = function setAAD(aadbuf, options) {
   if (!isArrayBufferView(aadbuf)) {
     throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'buffer',
                                ['Buffer', 'TypedArray', 'DataView']);
   }
-  if (this._handle.setAAD(aadbuf) === false)
+
+  let plaintextLength = -1;
+  if (options && options.plaintextLength != null) {
+    plaintextLength = options.plaintextLength;
+    if (~~plaintextLength !== plaintextLength || plaintextLength < 0)
```

Comment:
would ìf (plaintextLength >>> 0 !== plaintextLength)` also work?

Reply:
I think it should, let me try.